storm-core/src/jvm/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -3600,7 +3600,7 @@ public ComponentPageInfo getComponentPageInfo(String topoId, String componentId,
                 List<Integer> tasks = compToTasks.get(StormCommon.EVENTLOGGER_COMPONENT_ID);
                 tasks.sort(null);
                 // Find the task the events from this component route to.
-                int taskIndex = TupleUtils.listHashCode(Arrays.asList(componentId)) %
+                int taskIndex = (TupleUtils.listHashCode(Arrays.asList(componentId)) % tasks.size() + tasks.size()) %
```

Comment:
This should match the task selected by fields grouping https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/daemon/GrouperFactory.java#L159. 
The modulo technique will return a different value than Math.abs. Probably we should move it to some common utility function and use it in both places.

In 1.x branch the fields grouper doesn't seem to account for negative hashCode, so not sure why didn't it come up before.